src/unstract/sdk/ocr.py

```diff
@@ -8,47 +8,61 @@
 
 from unstract.sdk.adapters import ToolAdapter
 from unstract.sdk.constants import LogLevel
+from unstract.sdk.exceptions import OCRError
 from unstract.sdk.tool.base import BaseTool
 
 
 class OCR(metaclass=ABCMeta):
     def __init__(
         self,
         tool: BaseTool,
-        adapter_instance_id: str,
+        adapter_instance_id: str = None,
     ):
-        self.tool = tool
-        self.ocr_adapters = adapters
-        self.adapter_instance_id = adapter_instance_id
-        self.ocr_instance: OCRAdapter = self._get_ocr()
+        self._tool = tool
+        self._ocr_adapters = adapters
+        self._adapter_instance_id = adapter_instance_id
+        self._ocr_instance: OCRAdapter = None
+        self._initialise(adapter_instance_id)
+
+    def _initialise(self, adapter_instance_id):
+        if self._adapter_instance_id:
+            self._ocr_instance: OCRAdapter = self._get_ocr()
 
     def _get_ocr(self) -> Optional[OCRAdapter]:
         try:
+            if not self._adapter_instance_id:
+                raise OCRError("Adapter instance ID not set. " "Initialisation failed")
             ocr_config = ToolAdapter.get_adapter_config(
-                self.tool, self.adapter_instance_id
+                self._tool, self._adapter_instance_id
             )
             ocr_adapter_id = ocr_config.get(Common.ADAPTER_ID)
-            if ocr_adapter_id in self.ocr_adapters:
-                ocr_adapter = self.ocr_adapters[ocr_adapter_id][Common.METADATA][
+            if ocr_adapter_id in self._ocr_adapters:
+                ocr_adapter = self._ocr_adapters[ocr_adapter_id][Common.METADATA][
                     Common.ADAPTER
                 ]
                 ocr_metadata = ocr_config.get(Common.ADAPTER_METADATA)
-                self.ocr_instance = ocr_adapter(ocr_metadata)
+                self._ocr_instance = ocr_adapter(ocr_metadata)
 
-                return self.ocr_instance
+                return self._ocr_instance
 
         except Exception as e:
-            self.tool.stream_log(
-                log=f"Unable to get OCR adapter {self.adapter_instance_id}: {e}",
+            self._tool.stream_log(
+                log=f"Unable to get OCR adapter {self._adapter_instance_id}: {e}",
                 level=LogLevel.ERROR,
             )
             return None
 
     def process(
         self, input_file_path: str, output_file_path: Optional[str] = None
     ) -> str:
-        return self.ocr_instance.process(input_file_path, output_file_path)
+        return self._ocr_instance.process(input_file_path, output_file_path)
 
-    @deprecated("Use the class instance")
+    @deprecated(
+        "Deprecated method. Use the latest OCR to instantiate "
+        "and then invoke process()"
```

Comment:
@gaya3-zipstack `deprecated` library will throw warning that it is deprecated, we do not need to mention again.

In that case let's change message to `Use process() method of OCR instance.`.

Reply:
Done